src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1007,7 +1167,7 @@ public function has_class( $wanted_class ) {
 	 */
 	public function set_bookmark( $name ) {
 		// It only makes sense to set a bookmark if the parser has paused on a concrete token.
-		if ( self::STATE_MATCHED_TAG !== $this->parser_state ) {
+		if ( self::STATE_INCOMPLETE === $this->parser_state ) {
```

Comment:
I was looking at the states we can be in. Is it safe to say we should be in a state that is not the end of the document?

If so, should we also return false on `STATE_COMPLETE`?

```suggestion
		if ( self::STATE_COMPLETE === $this->parser_state || self::STATE_INCOMPLETE === $this->parser_state ) {
```

Reply:
Good catch. Added in 9f2992070ebbbb8970c96c115da2fb74a7ebfda4